drafts/two-way-binding-signals-and-query-params.md

```diff
@@ -0,0 +1,754 @@
+
+Wait a minute, really? Weren't those important APIs already available when the Angular team released the first stable version of signals? Isn't two-way binding or, even more, parent-to-child communication basic features of the framework? Yes, we could claim that but the Angular team is taking a gradual strategy to introduce all these new concepts into the framework. And that's probably a good thing, since they are learning and listening from the community during the process. You can also influence and give your opinion in the [open discussions][angularDiscussions].
+
+In today's article, we'll experiment and push to the limit the current state of signals in the Angular framework. For that, we will implement an exciting Two-way binding mechanism to connect signals with Angular's router state and the browser url. Are you ready? 
+
+## The Challenge
+
+Let's start with a simple application:
+
+```ts
+@Component({
+  selector: "app-root",
+  standalone: true,
+  template: ` <h1>Signals Demo</h1> `,
+})
+export class AppComponent {}
+```
+
+![Initial application](https://cdn.hashnode.com/res/hashnode/image/upload/v1710536390017/q5rnF-2Oc.png?auto=format&w=500)
+
+This is pretty much what you get when you generate a new Angular application nowadays. But without all the boilerplate and some basic styles.
+
+Let's start by consuming query params in the traditional way: using `RxJS`:
+
+```ts
+@Component({
+  selector: "app-root",
+  standalone: true,
+  imports: [AsyncPipe],
+  template: `
+    <h1>Signals Demo</h1>
+    <p>Your first name is: {{ firstName$ | async }}</p>
+  `,
+})
+export class AppComponent {
+  private activatedRoute = inject(ActivatedRoute);
+
+  firstName$ = this.activatedRoute.queryParams.pipe(
+    map((allQueryParams) => allQueryParams["firstName"])
+  );
+}
+```
+
+![Consuming query params with RxJS](https://cdn.hashnode.com/res/hashnode/image/upload/v1710536464769/FbnjIrApA.gif?auto=format)
+
+Great! this works and it's not necessarily inconvenient since Angular is handling the tedious task of managing subscriptions when we use the `async` pipe. Now let's see how this would look like with signals:
+
+At the moment of writing this article, Angular's router doesn't provide signals as results when we use its API. Most likely this will be introduced in future releases, but for now we have to do a workaround. We can pick the current responses from Angular's router, which are based on `RxJS`'s observables, and convert them by using the `toSignal` utility function:
+
+```ts
+@Component({
+  selector: "app-root",
+  standalone: true,
+  template: `
+    <h1>Signals Demo</h1>
+    <p>Your first name is: {{ firstName() }}</p>
+  `,
+})
+export class AppComponent {
+  private activatedRoute = inject(ActivatedRoute);
+
+  firstName = toSignal(
+    this.activatedRoute.queryParams.pipe(map((allQueryParams) => allQueryParams["firstName"]))
```

Comment:
**suggestion:** probably makes sense to put `map` on a new line again (like in the previous example), code in Hashnode is quite large and let's avoid horizontal scrolling as much as possible